src/merkle_tree/constraints.rs

```diff
@@ -1,96 +1,40 @@
+use crate::crh::TwoToOneCRHGadget;
+use crate::merkle_tree::Config;
+use crate::{CRHGadget, Path};
 use ark_ff::Field;
+use ark_r1cs_std::alloc::AllocVar;
+use ark_r1cs_std::boolean::Boolean;
+use ark_r1cs_std::eq::EqGadget;
+#[allow(unused)]
 use ark_r1cs_std::prelude::*;
+use ark_r1cs_std::ToBytesGadget;
 use ark_relations::r1cs::{Namespace, SynthesisError};
+use ark_std::borrow::Borrow;
 
-use crate::{
-    crh::{FixedLengthCRH, FixedLengthCRHGadget},
-    merkle_tree::*,
-};
-
-use core::borrow::Borrow;
-
-pub struct PathVar<P, HGadget, ConstraintF>
+/// Represents a merkle tree path gadget.
+pub struct PathVar<P, LeafHG, TwoToOneHG, ConstraintF>
 where
     P: Config,
-    HGadget: FixedLengthCRHGadget<P::H, ConstraintF>,
+    LeafHG: CRHGadget<P::LeafHash, ConstraintF>,
+    TwoToOneHG: TwoToOneCRHGadget<P::TwoToOneHash, ConstraintF>,
```

Comment:
Thanks for updating the `HG` to `H` suffix in the impl! Still needs to be done for the struct definition =)

Reply:
Oops! My bad

Let me look if there's any other `HG` there